### PR TITLE
Make ctrl-click open new tab

### PR DESCRIPTION
**Pull request by theFeu**
refs #3722

### CODE REVIEW

File: public/js/icinga/events.js [hunk 1]
```diff
@@ -457,6 +457,12 @@
                 return true;
             }
 
+            // Check for ctrl or cmd click to open new tab
+            if (event.ctrlKey || event.metaKey) {
```
Comment (nilmerg) [inline public/js/icinga/events.js]: What's the `metaKey`? Is this an apple specific thing?

Comment (theFeu) [reply public/js/icinga/events.js]: Form the [MDN web docs](https://developer.mozilla.org/en-US/docs/Web/API/KeyboardEvent/metaKey)
On Macintosh keyboards, this is the ⌘ Command key. On Windows keyboards, this is the Windows key (⊞ Windows).

**Comment by theFeu** [comment]
> Works very well in the menu. Though, list entries behave weird now. They open in a new tab in addition to the detail view. It also interferes with the selection of multiple rows.

I'll look into this ASAP :)

**Comment by theFeu** [comment]
@nilmerg 
What behaviour would you expect on clicking the list-item?
A new browser tab opening but no detail view?
No tab opening?

This interferes with it being possible to make multiselects with the ctrl/cmd key.

So I see two possibilities on how to handle this: we keep the behaviour as it is (normal multiselect + new tab opeing) or we remove the crtl/cmd keys from triggering multiselect.

**Comment by nilmerg** [comment]
I generally only expect the ctrl+click behavior to work when directly aiming at an anchor. Doing this on text or empty areas I wouldn't expect this to work. To select multiple elements I rather tend to click anywhere but the anchor.

**Comment by theFeu** [comment]
should be fixed now :)

**Comment by nilmerg** [comment]
Are you sure? Shows the same behavior for me.

**Comment by theFeu** [comment]
Fixed me being an idiot too now ^-^'

**Comment by nilmerg** [comment]
Yep, the host list and services list work fine even with multiselection. Though, some lists still open in a new tab *and* open a column. (e.g. `/icingaweb2/user/list`)

**Comment by theFeu** [comment]
> Yep, the host list and services list work fine even with multiselection. Though, some lists still open in a new tab and open a column. (e.g. /icingaweb2/user/list)

And you think it should not open a new column?
Or no new tab?

Since the multiselect is not possible here, I don't see the problem with it opening a new tab - even when clicking on the row instead of the anchor.
The meta key does not serve any other function in this (except for open new tab, that is).

Plus: I personally really like having a little preview over what i just opened in a new tab :)

**Comment by nilmerg** [comment]
Well, the default behavior of a browser in this case is to open the target in a new window (for browsers nowadays in a new tab) while the page where the link was clicked stays the same. Consider for a moment someone clicking on a link like this in the right-most column. The left column is lost, what's probably not what the user wanted.

**Comment by theFeu** [comment]
> The left column is lost, what's probably not what the user wanted.

This is something I definitely didn't think about!

File: public/js/icinga/events.js [hunk 2]
```diff
@@ -457,10 +457,16 @@
                 return true;
             }
 
+            // Check for ctrl or cmd click to open new tab unless clicking on a multiselect row
+            if ((event.ctrlKey || event.metaKey) && ! $a.parent().parent('table').length > 0 && href !== '#') {
```
Comment (nilmerg) [inline public/js/icinga/events.js]: Please use the same condition as on line no. 500 to check for multiselect tables. (And outsource this to a variable for re-use) The current condition also matches the tactical overview's fake-links and still opens a new column due to this.

Comment (theFeu) [reply public/js/icinga/events.js]: If I use the multiselect condition other tables and lists (e.g. the `/icingaweb2/user/list` that you mentioned will no longer work as intended.
The tactical overview does not really fit in with the way we display in other views in general - maybe that should change?

Comment (nilmerg) [reply public/js/icinga/events.js]: The condition I've mentioned also verifies that it's a `tr` tag. Links shouldn't be affected by this.

Comment (theFeu) [reply public/js/icinga/events.js]: So, what if instead i just verify that it is an anchor?
Just `(event.ctrlKey || event.metaKey) && $a.is('a') && href !== '#'`

Comment (nilmerg) [reply public/js/icinga/events.js]: That's also fine. Though, then I'd rather want to see the check at the very end. The `$a.is('a')` is probably the most expensive one.

**Comment by theFeu** [comment]
The Tactical overview would also be less of a special case if we would change the legend to behave like other 'badge collections'.
See: #3797

**Pull request merged**